dart/lib/src/http_client/sentry_http_client.dart

```diff
@@ -43,57 +44,22 @@ import '../hub_adapter.dart';
 /// }
 /// ```
 class SentryHttpClient extends BaseClient {
```

Comment:
I've splitted `SentryHttpClient` into `FailedRequestClient` and `BreadcrumbClient`. This makes it easier to test and also easier to add more functionality or to remove functionality. Also the user can choose to only use the ones he needs.

I still need to make `SentryHttpClient` configurable so that `FailedRequestClient` is opt-in by default.

Reply:
mmm, I'm wondering how this would work, today people add `SentryHttpClient`, does this change?
how I opt in only for failed requests or breadcrumbs? how do I use both?
I imagined this with a sort of optional ctor params for `SentryHttpClient`, I need to check how friendly this would be for the end user now

Reply:
one thing to consider is, both clients call `send`, does it mean it executes the URL request twice? another thing to consider is, if the 1st client rethrows, does the other client still executes correctly? I guess spitting up may create a bit more complexity than making it easier to test, we can discuss this offline

Reply:
also, this is a breaking change, setting `SentryHttpClient` which is the latest stable version would not capture errored HTTP calls, now it will, and its a breaking change, ideally those features are opt-in and not opt-out, till we get it stable and do a major bump

Reply:
Side note: What I've done here is to apply the decorator pattern. Each client adds functionality. Each decorator just calls its inner object and the innermost object is a Client (either supplied by the user or created by us).